hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/multitenant/RangerClientMultiTenantAccessController.java

```diff
@@ -130,9 +130,15 @@ public RangerClientMultiTenantAccessController(OzoneConfiguration conf)
 
     LOG.info("authType = {}, login user = {}", authType, usernameOrPrincipal);
 
-    client = new RangerClient(rangerHttpsAddress,
-        authType, usernameOrPrincipal, passwordOrKeytab,
-        rangerServiceName, OzoneConsts.OZONE);
+    UserGroupInformation loginUser = UserGroupInformation.getLoginUser();
+    try {
+      client = new RangerClient(rangerHttpsAddress,
+          authType, usernameOrPrincipal, passwordOrKeytab,
+          rangerServiceName, OzoneConsts.OZONE);
+    } finally {
+      // set back the expected login user
+      UserGroupInformation.setLoginUser(loginUser);
+    }
```

Comment:
Thanks Sammi. Good catch.

From what I understand of HDDS-11587 description, `RangerClient` shares the same `UserGroupInformation` context with its caller (OM in this case).

Thus, when `RangerClient`'s internal Kerberos login logic generated a new UGI and replaces the caller's LoginUser UGI, which causes the breakage (OM stuck, failed to re-login after a while).

Questions for Ranger folks: @kumaab
1. What happens when `RangerClient`'s UGI Kerberos login expires as well? My concern is, if `RangerClient` re-login using similar logic, the `RangerClient` instance can get stuck for the same reason as well? Or if `RangerClient` sets UGI again (haven't checked the impl), the OM could get stuck again for the same reason.
3. Is there a way to isolate `RangerClient`'s internal `UserGroupInformation` so that it doesn't leak out? Or, is it feasible to share the same UGI with caller?

Reply:
There is a "RangerTagRefresher" thread which will periodically do kerberos relogin before ticket expiry. It doesn't call setLoginUser(). So OM will not stuck again. 
Actually I checked two different Ranger versions, in one version setLoginUser is not called during RangerClient(). I'm curious for what purpose this setLoginUser is introduced.